### DISCUSSION TITLE
Verifying Supported Platforms

### DISCUSSION THREAD

**Discussion by wiemerm** [wwt/SwiftCurrent#90] [Spikes and Findings]
Verifying platforms other than iOS are now supported with the introduction of SwiftUI support. 

For SPM, changed the `Package.swift` file to include macOS, watchOS, and tvOS in the supported platforms.

`    platforms: [.iOS(.v11), .macOS(.v11), .tvOS(.v14), .watchOS(.v7)],
`
Versions for macOS, tvOS, and watchOS were based off of our `@available`

Starting with a simple iOS project, we added new targets for macOS, watchOS, and tvOS respectively and added the files from the iOS target to also be part of the other targets. macOS was the only target where the `WorkflowLauncher` code was duplicated into its own targets' `ContentView` (whereas tvOS and watchOS deleted the `ContentView` files, and that only to make it look a bit nicer on macOS.

Using `WorkflowLauncher` just worked across the various platforms for a simple workflow.

**Comment by morganzellers** [comment]
For CocoaPods, we've been toying around with our PodSpec file using the localpod/devpod approach. 

Thus far, the SwiftUI Beta is pulling in on watchOS and tvOS with some routine changes to the `podspec`. macOS I'm expecting will go easily as well. I currently have these set up as separate projects importing SwiftCurrent into the `ContentView` files. Planning to consolidate these and go through a 3 node workflow on each as well as checking the UIKit support where appropriate.

**Reply by morganzellers** [reply]
Status update:

CocoaPods & SwiftUI:

- tvOS ✅
  - Able to run the SwiftUI getting started
- watchOS
  - Line 13 of `TestOnly.swift` was throwing build errors due to `XCTest` not being supported in Xcode 12.4
  - (Feeling like we don't want that to get packed into the product, but I may be mistaken)
  - Pod was pulled in fine otherwise 
- macOS
  - Was not able to test out the SwiftUI approach due to not being on Big Sur/macOS 11
  - Pod was pulled in fine otherwise

**Reply by Tyler-Keith-Thompson** [reply]
We do want `TestOnly.swift` in the product since it allows for better testing kinds of things. It's really weird that the active compilation condition is causing issues, that's sort of its point, if there is no XCTest don't worry about the code. 

This should work just fine with 12.5 since watchOS now supports XCTest but 12.4 and below may always have issues, lame.

**Comment by morganzellers** [comment]
Proposed changes to the `.podspec` file:

~**Add to the SwiftCurrent spec:**~

~`s.macos.deployment_target = '11.0'`~
~`s.watchos.deployment_target = '7.0'`~
~`s.tvos.deployment_target = '14.0'`~
  
**Add to the Core subspec:**

`ss.watchos.deployment_target = '7.0'`
`ss.tvos.deployment_target = '14.0'`
`ss.ios.deployment_target = '11.0'`
`ss.macos.deployment_target = '11.0'`


**Add to the BETA_SwiftUI subspec:**

`ss.watchos.deployment_target = '7.0'`
`ss.tvos.deployment_target = '14.0'`
`ss.ios.deployment_target = '11.0'`
`ss.macos.deployment_target = '11.0'`

**Reply by Richard-Gist** [reply]
I agree with Core and BETA.  We could probably roll Core back even further, but this looks good enough.  I'm confused by:
> Add to the SwiftCurrent spec:

Is that referring to the whole package?  Because you can't have watchos with UIKit, nor macOS.  I'm unsure if tvOS has UIKit either.

**Reply by morganzellers** [reply]
On review, we don't need those! I'll edit.

**Comment by Richard-Gist** [comment]
A PR has been created #94 to do these changes
